EngineLayer/Silac/SilacConversions.cs

```diff
@@ -332,267 +269,602 @@ public static SpectraFileInfo GetHeavyFileInfo(SpectraFileInfo originalFile, Sil
-                                    );
-                                identifications.Add(updatedId);
+                                //overwrite the first three peptide quantification values with the total values shown here IF there was more than one missed cleavage
+                                for (int i = 0; i < 3; i++)
+                                {
+                                    peptides[i].SetIntensity(info, values[i]);
+                                }
                             }
-                            FlashLFQ.ChromatographicPeak updatedPeak = new FlashLFQ.ChromatographicPeak(identifications.First(), peak.IsMbrPeak, peak.SpectraFileInfo);
-                            for (int j = 1; j < identifications.Count; j++) //add all the original identification
+                            //else if three values, then we're fine
+                            //add the values from this peptide to the file-specific-values of all peptides
+                            double[] totalValues = fileToRecycleDictionary[info];
+                            for (int i = 0; i < values.Count; i++)
                             {
-                                updatedPeak.MergeFeatureWith(new FlashLFQ.ChromatographicPeak(identifications[j], peak.IsMbrPeak, peak.SpectraFileInfo), Integrate);
+                                totalValues[i] += values[i];
                             }
-                            updatedPeak.IsotopicEnvelopes = peak.IsotopicEnvelopes; //need to set isotopicEnevelopes, since the new identifications didn't have them.
-                            updatedPeak.CalculateIntensityForThisFeature(Integrate); //needed to update info
-                            peaks[i] = updatedPeak;
                         }
                     }
-                }
 
-                //convert all lfq peptides from heavy to light for output
-                Dictionary<string, FlashLFQ.Peptide> lfqPwsms = FlashLfqResults.PeptideModifiedSequences;
-                List<string> pwsmKeys = lfqPwsms.Keys.ToList();
-                foreach (string key in pwsmKeys)
-                {
-                    FlashLFQ.Peptide currentPeptide = lfqPwsms[key];
-                    SilacLabel label = GetRelevantLabelFromFullSequence(currentPeptide.Sequence, silacLabels);
-                    if (label != null) //if it's a heavy peptide
+                    //with the summed LL, LH, and HH values for each file, we can calculate the probability that a heavy amino acid is incorporated (Ph)
+                    //the probability a light amino acid is incorporated (Pl) is simply Pl = 1 - Ph
+                    //We can simplify the math if we assume that all protein started as completely light, @t=0: LL=100%, LH = 0%, and HH = 0%
+                    //N = 1 - Qll + Qlh * Qlh / (4 * Qhh), (where N is the fraction of all protein that has turned over and Qll is the Quantity(fraction) of LL at the current timepoint
```

Comment:
is it defaulted to "light" or whatever the start condition is? don't you go from heavy to light in one of your experiments?

Reply:
The comments use L/H for ease of understanding (because typical experiments move from L->H), but they're a proxy for start/end, which is what is actually used in the code.